packages/core/src/entity/EntityStoreBase.ts

```diff
@@ -27,6 +30,78 @@ export class EntityStoreBase {
     this.addEntity(entity);
   }
 
+  public getValueDeep(
+    headLinkOrEntity: UnresolvedLink<'Entry' | 'Asset'>,
+    deepPath: string
+  ): string | undefined {
+    const resolveFieldset = (
+      unresolvedFieldset: Array<[null, string, string?]>,
+      headEntity: Entry | Asset
+    ) => {
+      const resolvedFieldset: Array<[Entry | Asset, string, string?]> = [];
+      let entityToResolveFieldsFrom: Entry | Asset = headEntity;
+      for (let i = 0; i < unresolvedFieldset.length; i++) {
+        const isLeaf = i === unresolvedFieldset.length - 1; // with last row, we are not expecting a link, but a value
+        const row = unresolvedFieldset[i];
+        const [_, field, _localeQualifier] = row;
+        if (!entityToResolveFieldsFrom) {
+          throw new Error(
+            `Logic Error: Cannot resolve field ${field} of a fieldset as there is no entity to resolve it from.`
+          );
+        }
+        if (isLeaf) {
+          resolvedFieldset.push([entityToResolveFieldsFrom, field, _localeQualifier]);
+          return resolvedFieldset;
+        }
+
+        const fieldValue = get<string>(entityToResolveFieldsFrom, ['fields', field]) as
+          | UnresolvedLink<'Entry' | 'Asset'>
+          | undefined
+          | any;
+
+        if (undefined === fieldValue) {
+          throw new Error(`Cannot resolve field ${field} of a fieldset as it is not defined.`);
+        } else if (isLink(fieldValue)) {
+          let entity: Asset | Entry | undefined = this.getEntityFromLink(fieldValue);
```

Comment:
```suggestion
          let entity = this.getEntityFromLink(fieldValue);
```
I'd suggest you set a return type in the definition of getEntityFromLink  to avoid having to explicitly set it here:
```
public getEntityFromLink(link: UnresolvedLink<'Entry' | 'Asset'>): Asset | Entry | undefined {
```

Reply:
Done!